PR: Changed issue numbers in code to links

oscargus opened the pull request:
<!--- Make sure to read the Contributing Guidelines:                   --->
<!--- https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md --->
<!--- and follow PEP 8, PEP 257 and Spyder's code style:               --->
<!--- https://github.com/spyder-ide/spyder/wiki/Dev:-Coding-Style      --->

## Description of Changes
Inspired by https://github.com/spyder-ide/spyder/pull/9710#discussion_r298670806 I went ahead and changed all issue numbers I could find in the code... (except for the file in #9710)

<!--- Explain what you've done and why --->





### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct: Oscar Gustafsson/oscargus

<!--- Thanks for your help making Spyder better for everyone! --->

ccordoba12 commented on spyder/app/tests/test_mainwindow.py:
```diff
@@ -1688,7 +1688,8 @@ def test_troubleshooting_menu_item_and_url(monkeypatch):
 def test_help_opens_when_show_tutorial_full(main_window, qtbot):
-    """Test fix for #6317 : 'Show tutorial' opens the help plugin if closed."""
+    """Test fix for spyder-ide/spyder#6317 :
+    'Show tutorial' opens the help plugin if closed."""
```
This is incorrectly formatted. It needs to be

```python
"""
Test fix for spyder-ide/spyder#6317 :

'Show tutorial' opens the help plugin if closed.
"""
```

oscargus replied:
OK! So never text on a line with """? When there is more than one line?

Btw, should it be a space after the `spyder-ide/spyder#XXXX`? In the code it varies. (I assume it can be a space as here, but is it better with a space?)

ccordoba12 replied:
> OK! So never text on a line with """? When there is more than one line?

If your docstring doesn't fit in a single line (i.e. if it has more than 79 chars), then you need to split it in a first line that contains a summary and subsequent lines (separated by a blank line) that contain a description of the function/method/class being documented.

> Btw, should it be a space after the spyder-ide/spyder#XXXX?

Sorry, I should have suggested you to remove the colon too. But that's not much of an issue.

ccordoba12 replied:
You can remove that colon if you want.

pep8speaks commented:
Hello @oscargus! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







































































* In the file [`spyder/plugins/ipythonconsole/widgets/client.py`](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/ipythonconsole/widgets/client.py):

> [Line 549:80](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/ipythonconsole/widgets/client.py#L549): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)























* In the file [`spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py`](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py):

> [Line 199:1](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py#L199): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 221:1](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py#L221): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 227:1](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py#L227): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 227:43](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py#L227): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment





* In the file [`spyder/utils/bsdsocket.py`](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/utils/bsdsocket.py):

> [Line 113:1](https://github.com/spyder-ide/spyder/blob/2d6c6c81748d1218c65ad4592a357b38363b5c85/spyder/utils/bsdsocket.py#L113): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment

oscargus commented:
Fixed the comment.

For the empty lines, I think that they are simply removed spaces. I have that turned on in Spyder as it is quite beneficial in almost all other cases (I guess pep8speaks will complain if I accidentally added them on an edited line). I see the possible merge problems though.

Now, in the last commit, I added some empty lines as pep8speaks complained. (As I updated the comments on/near these lines.)

ccordoba12 commented:
@oscargus, please address the comments left by pep8speaks so I can merge this one.

ccordoba12 commented:
And you also need to merge with master now to fix the merge conflict that appeared.

goanpeca commented on spyder/workers/updates.py:
```diff
@@ -94,7 +94,8 @@ def start(self):
         try:
             if hasattr(ssl, '_create_unverified_context'):
-                # Fix for issue # 2685 [Works only with Python >=2.7.9]
+                # Fix for # See spyder-ide/spyder#90442685
```
@oscargus is this a typo?

Was it like this?

`See spyder-ide/spyder#2685`

oscargus replied:
Not sure what happened there...

goanpeca commented on spyder/widgets/tabs.py:
```diff
@@ -474,6 +474,6 @@ def move_tab_from_another_tabwidget(self, tabwidget_from,
         # depend on the platform: long for 64bit, int for 32bit. Replacing
         # by long all the time is not working on some 32bit platforms
-        # (see Issue 1094, Issue 1098)
+        # (see spyder-ide/spyder#1094, spyder-ide/spyder#1098)
```
see -> See

goanpeca commented on spyder/widgets/tabs.py:
```diff
@@ -218,7 +218,7 @@ def dropEvent(self, event):
             # depend on the platform: long for 64bit, int for 32bit. Replacing
             # by long all the time is not working on some 32bit platforms
-            # (see Issue 1094, Issue 1098)
+            # (see spyder-ide/spyder#1094, spyder-ide/spyder#1098)
```
see -> See

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py:
```diff
@@ -544,7 +549,7 @@ def test_dataframeeditor_edit_complex(qtbot, monkeypatch):
     Test that editing complex dtypes is handled gracefully in df editor.
 
-    Integration regression test for issue #6115 .
+    Integration regression test for spyder-ide/spyder#6115 .
```
Although not related to the actual change, this texts sounds a bit odd.

Maybe 

```
Integration and regression test for spyder-ide/spyder#6115
```

or
```
Integration test for regression spyder-ide/spyder#6115
```

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:
```diff
@@ -390,7 +392,7 @@ def test_xml_dom_element_view():
     Test that XML DOM ``Element``s are able to be viewied in CollectionsEditor.
 
-    Regression test for issue #5642 .
+    Regression test for spyder-ide/spyder#5642 .
```
I think we can either remove the dot `.` or do not leave a space?

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:
```diff
@@ -410,7 +412,7 @@ def test_pandas_dateoffset_view():
     Test that pandas ``DateOffset`` objs can be viewied in CollectionsEditor.
 
-    Regression test for issue #6729 .
+    Regression test for spyder-ide/spyder#6729 .
```
I think we can either remove the dot `.` or do not leave a space?

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:
```diff
@@ -317,7 +319,7 @@ def test_notimplementederror_multiindex():
     Test that the NotImplementedError when scrolling a MultiIndex is handled.
 
-    Regression test for issue #6284 .
+    Regression test for spyder-ide/spyder#6284 .
```
I think we can either remove the dot `.` or do not leave a space?

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:
```diff
@@ -208,7 +209,7 @@ def test_edit_mutable_and_immutable_types(monkeypatch):
     Test that mutable objs/vals are editable in VarExp; immutable ones aren't.
 
-    Regression test for issue #5991 .
+    Regression test for spyder-ide/spyder#5991 .
```
I think we can either remove the dot `.` or do not leave a space?

goanpeca replied:
I prefer just removing the dot (so it reads cleaner :-)  )

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:
```diff
@@ -307,7 +308,8 @@ def test_view_module_in_coledit():
 
-    Also check that they are set as readonly. Regression test for issue #6080 .
+    Also check that they are set as readonly. Regression test for
+    spyder-ide/spyder#6080 .
```
I think we can either remove the dot `.` or do not leave a space?

goanpeca commented on spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:
```diff
@@ -337,7 +339,7 @@ def test_editor_parent_set(monkeypatch):
     Test that editors have their parent set so they close with Spyder.
 
-    Regression test for issue #5696 .
+    Regression test for spyder-ide/spyder#5696 .
```
I think we can either remove the dot `.` or do not leave a space?

goanpeca commented on spyder/plugins/ipythonconsole/widgets/client.py:
```diff
@@ -546,7 +547,7 @@ def restart_kernel(self):
                     )
                 else:
-                    # For issue 6235.  IPython was changing the setting of
+                    # For spyder-ide/spyder#6235.  IPython was changing the setting of
```
For spyder-ide/spyder#6235.  IPython was changing the setting of ->

For spyder-ide/spyder#6235. IPython was changing the setting of

(remove an extra space between 6235.  IPython)

goanpeca commented on spyder/app/tests/test_mainwindow.py:
```diff
@@ -1688,7 +1688,11 @@ def test_troubleshooting_menu_item_and_url(monkeypatch):
 def test_help_opens_when_show_tutorial_full(main_window, qtbot):
-    """Test fix for #6317 : 'Show tutorial' opens the help plugin if closed."""
+    """
+    Test fix for spyder-ide/spyder#6317 :
```
I think we should settle on a style, and try to use the same style for all inside docstrings.

Since the regex will work well no matter if a dot or colon is added, I guess the best would be to always use periods `.` within docstrings, if the spyder-ide/spyder#NNNN is at the end of a sentence/line

goanpeca commented on spyder/app/mainwindow.py:
```diff
@@ -3022,7 +3022,7 @@ def start_open_files_server(self):
                 req, dummy = self.open_files_server.accept()
             except socket.error as e:
-                # See Issue 1275 for details on why errno EINTR is
+                # Seespyder-ide/spyder#1275 for details on why errno EINTR is
```
Seespyder-ide/spyder#1275 -> See spyder-ide/spyder#1275

oscargus commented:
(Been away on vacation...)

@ccordoba12 : I think most of them are fixed, but pep8speaks didn't update. New attempt when sorting out the comments from @goanpeca though.

@goanpeca : will do! Did a bit of automatic find-replace after a while, so some things are not that well checked...

oscargus commented:
I've updated it and I think I have managed to add periods etc in the correct places (and found #9765).

ccordoba12 commented:
Thanks @oscargus for addressing our last comments.

@goanpeca, please review again and give your final approve, if this is ready.
